EventGenerator/src/MuCapGammaRayGenerator_tool.cc

```diff
@@ -0,0 +1,65 @@
+#include "art/Utilities/ToolMacros.h"
+
+#include "CLHEP/Random/RandFlat.h"
+
+#include "EventGenerator/inc/ParticleGeneratorTool.hh"
+
+#include "DataProducts/inc/PDGCode.hh"
+#include "MCDataProducts/inc/GenId.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+#include "Mu2eUtilities/inc/BinnedSpectrum.hh"
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+
+namespace mu2e {
+  class MuCapGammaRayGenerator : public ParticleGeneratorTool {
+  public:
+    explicit MuCapGammaRayGenerator(Parameters const& conf) :
+      _pdgId(PDGCode::gamma),
+      _mass(GlobalConstantsHandle<ParticleDataTable>()->particle(_pdgId).ref().mass().value()),
+      _genId(GenId::MuCapGammaRayGenTool),
+      _energy(GlobalConstantsHandle<PhysicsParams>()->getCaptureGammaEnergy()),
+      _intensity(GlobalConstantsHandle<PhysicsParams>()->getCaptureGammaIntensity())
+    {
+
+    }
+      void generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) override;
+
+    void setEngine(art::RandomNumberGenerator::base_engine_t& eng) {
+      _randomUnitSphere = new RandomUnitSphere(eng);
+      _randFlat = new CLHEP::RandFlat(eng);
+    }
+
+  private:
+    PDGCode::type _pdgId;
+    double _mass;
+    GenId _genId;
+    double _energy;
+    double _intensity;
+
+    SpectrumVar       _spectrumVariable;
+    BinnedSpectrum    _spectrum;
```

Comment:
Please remove unused _spectrumVariable and _spectrum.    It would also make sense to do away wiith _mass for the photon generator.

Reply:
Done in commit 6af04d720 (along with generator tool renaming)